### PR TITLE
Add Trino 450 release notes

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 450 release.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 1 June 2024

* #22225 ✅ rn ✅ docs
* #22224 ✅ rn ✅ docs
* #22223 ✅ rn ✅ docs
* #22226 ✅ rn ✅ docs

## 2 June 2024

* #22228 ✅ rn ✅ docs
* #22230 ✅ rn ✅ docs
* #22234 ✅ rn ✅ docs
* #22236 ✅ rn ✅ docs

## 3 June 2024

* #22212 ✅ rn ✅ docs
* #22237 ✅ rn ✅ docs
* #22239 ✅ rn ✅ docs
* #22238 ✅ rn ✅ docs
* #22242 ✅ rn ✅ docs
* #22235 ✅ rn ✅ docs
* #21940 ✅ rn ✅ docs
* #22092 ✅ rn ✅ docs
* #22105 ✅ rn ❌ docs
* #22245 ✅ rn ✅ docs
* #22243 ✅ rn ✅ docs
* #22251 ✅ rn ✅ docs
* #22252 ✅ rn ✅ docs
* #22253 ✅ rn ✅ docs

## 4 June 2024

* #22241 ✅ rn ✅ docs
* #22039 ✅ rn ✅ docs
* #22257 ✅ rn ✅ docs
* #21727 ✅ rn ✅ docs
* #22266 ✅ rn ✅ docs
* #22269 ✅ rn ✅ docs
* #22263 ✅ rn ✅ docs
* #22265 ✅ rn ✅ docs
* #22255 ✅ rn ✅ docs

## 5 June 2024

* #22049 ✅ rn ✅ docs
* #22272 ✅ rn ✅ docs
* #22275 ✅ rn ✅ docs
* #22214 ✅ rn ✅ docs
* #21375 ✅ rn ❌ docs
* #22246 ✅ rn ✅ docs

## 6 June 2024

* #22283 ✅ rn ✅ docs
* #22277 ✅ rn ✅ docs
* #22295 ✅ rn ✅ docs
* #21988 ✅ rn ✅ docs
* #22290 ✅ rn ✅ docs
* #22301 ✅ rn ✅ docs
* #22262 ✅ rn ✅ docs
* #21875 ✅ rn ✅ docs
* #22291 ✅ rn ✅ docs
* #21904 ✅ rn ✅ docs
* #22298 ✅ rn ✅ docs
* #20866 ✅ rn ❌ docs
* #21663 ✅ rn ✅ docs

## 7 June 2024

* #22307 ✅ rn ✅ docs
* #22305 ✅ rn ✅ docs
* #22292 ✅ rn ✅ docs
* #22293 ✅ rn ✅ docs
* #22313 ✅ rn ✅ docs
* #22299 ✅ rn ✅ docs
* #22166 ✅ rn ✅ docs
* #22286 ✅ rn ✅ docs
* #22287 ✅ rn ✅ docs

## 8 June 2024

* #22338 ✅ rn ✅ docs

## 9 June 2024

* #22333 ✅ rn ✅ docs
* #22334 ✅ rn ✅ docs
* #22339 ✅ rn ✅ docs
* #22330 ✅ rn ✅ docs
* #22311 ✅ rn ✅ docs
* #22310 ✅ rn ✅ docs
* #22250 ✅ rn ✅ docs
* #22336 ✅ rn ✅ docs

## 10 June 2024

* #21907 ✅ rn ✅ docs
* #22346 ✅ rn ✅ docs
* #22345 ✅ rn ✅ docs
* #22343 ✅ rn ✅ docs
* #22337 ✅ rn ✅ docs
* #21773 ✅ rn ✅ docs
* #22340 ✅ rn ✅ docs

## 11 June 2024

* #22174 ✅ rn ✅ docs
* #22279 ✅ rn ✅ docs
* #22325 ✅ rn ✅ docs
* #22190 ✅ rn ✅ docs
* #22328 ✅ rn ✅ docs
* #22162 ✅ rn ✅ docs
* #22163 ✅ rn ✅ docs
* #22270 ✅ rn ✅ docs

## 12 June 2024

* #22363 ✅ rn ✅ docs
* #22348 ✅ rn ✅ docs
* #22364 ✅ rn ✅ docs
* #22209 ✅ rn ✅ docs
* #22320 ✅ rn ✅ docs
* #22296 ✅ rn ✅ docs
* #22354 ✅ rn ✅ docs
* #22218 ✅ rn ✅ docs
* #22367 ✅ rn ✅ docs
* #22159 ✅ rn ✅ docs
* #22335 ✅ rn ✅ docs
* #22142 ✅ rn ✅ docs
* #22365 ✅ rn ✅ docs
* #22372 ✅ rn ✅ docs
* #22215 ✅ rn ✅ docs
* #22369 ✅ rn ✅ docs

## 13 June 2024

* #22302 ✅ rn ✅ docs
* #22360 ✅ rn ✅ docs
* #22380 ✅ rn ✅ docs
* #21679 ✅ rn ✅ docs
* #22378 ✅ rn ✅ docs
* #22370 ✅ rn ✅ docs

## 14 June 2024

* #22396 ✅ rn ✅ docs
* #22395 ✅ rn ✅ docs
* #22350 ✅ rn ✅ docs

## 16 June 2024

* #22389 ✅ rn ✅ docs
* #22400 ✅ rn ✅ docs
* #22401 ✅ rn ✅ docs

## 17 June 2024

* #22402 ✅ rn ✅ docs
* #22192 ✅ rn ✅ docs
* #22383 ✅ rn ✅ docs
* #22294 ✅ rn ✅ docs
* #22405 ✅ rn ✅ docs
* #22382 ✅ rn ✅ docs

## 18 June 2024

* #22217 ✅ rn ✅ docs
* #22314 ✅ rn ✅ docs
* #22418 ✅ rn ✅ docs
* #22414 ✅ rn ✅ docs
* #22409 

## 19 June 2024

* #22417 
* #22427 
* #22435 
* #22446 
* #22439

### CODE REVIEW

File: docs/src/main/sphinx/release/release-450.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Release 450 (7 Jun 2024)
+
+## General
+
+* Add support for the `optimizer.push-filter-into-values-max-row-count`
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:5]: Invert 

Add support to define the number of rows to push from a filter into a value with the xx prop

File: docs/src/main/sphinx/release/release-450.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  configuration property, which can define a number of rows to be pushed from a
+  filter into values. ({issue}`22105`)
+* Improve performance of the {func}`first_value` and {func}`last_value`
+  functions. ({issue}`22092`)
+* Improve performance for large clusters under heavy workloads. ({issue}`22039`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:10]: super vague ..

File: docs/src/main/sphinx/release/release-450.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+* Improve performance of queries with simple predicates. This optimization can
+  be disabled using the `experimental.columnar-filter-evaluation.enabled`
+  configuration property or `columnar_filter_evaluation_enabled` session
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:13]: or the

File: docs/src/main/sphinx/release/release-450.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+## General
+
+* Add support for the `optimizer.push-filter-into-values-max-row-count`
+  configuration property, which can define a number of rows to be pushed from a
+  filter into values. ({issue}`22105`)
+* Improve performance of the {func}`first_value` and {func}`last_value`
+  functions. ({issue}`22092`)
+* Improve performance for large clusters under heavy workloads. ({issue}`22039`)
+* Improve performance of queries with simple predicates. This optimization can
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:11]: is "simple predicates" specific enough? I kinda think yes and people can look up more if needed. Also .. is this documented? And "experimental" .. do we need to mention that besides the name of the property?

File: docs/src/main/sphinx/release/release-450.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  be disabled using the `experimental.columnar-filter-evaluation.enabled`
+  configuration property or `columnar_filter_evaluation_enabled` session
+  property. ({issue}`21375`)
+* Fix failure when loading the {doc}`/admin/event-listeners-openlineage`. ({issue}`22228`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:15]: use markdown syntax 

```suggestion
* Fix failure when loading the [](/admin/event-listeners-openlineage). ({issue}`22228`)
```

File: docs/src/main/sphinx/release/release-450.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+## JDBC driver
+
+* Add support for the `assumeNullCatalogMeansCurrent` connection property. When
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:19]: fyi .. this is not documented on purpose since it seems to be super niche problem

File: docs/src/main/sphinx/release/release-450.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  enabled, a `null` value for the `catalog` parameter in `DatabaseMetaData`
+  methods is assumed to mean the current catalog. If no current catalog is
+  set, the behaviour is unmodified. ({issue}`20866`)
+
+## BigQuery connector
+
+* Add support for caching when the `bigquery.case-insensitive-name-matching`
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:26]: caching what?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-450.md:26]: and how is that related to the property?

File: docs/src/main/sphinx/release/release-450.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+* Add support for `integer` to `varchar` and `decimal` to `varchar` type 
+  coercion in unpartitioned tables. ({issue}`22246`, {issue}`22293`)
+* Add support for `double` to `varchar` type coercion in unpartitioned tables
+  using Parquet files. ({issue}`22277`)
+* Add support for `float` to `varchar` type coercion. ({issue}`22291`)
+
+## MariaDB connector
+
+* Improve performance of listing table columns. ({issue}`22241`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:50]: This does not affects SQL Server, PostgreSQL or Oracle?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-450.md (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-450.md:10]: I'm not really sure how else to word it...

File: docs/src/main/sphinx/release/release-450.md (hunk 8 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-450.md:50]: https://github.com/trinodb/trino/pull/22241 seems to only list those four

Comment (mosabua) [reply docs/src/main/sphinx/release/release-450.md:50]: ok

File: docs/src/main/sphinx/release/release-450.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-450.md:10]: yeah .. fine to leave as is

**Review by mosabua — APPROVED** [review]
Seems good to me now.

File: docs/src/main/sphinx/release/release-450.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
+# Release 450 (7 Jun 2024)
+
+## General
+
+* Add support for defining a number of rows to be pushed from a filter into
```
Comment (martint) [inline docs/src/main/sphinx/release/release-450.md:5]: This should describe the user-visible impact. "pushed from a filter into values" doesn't mean anything for an end user.

Comment (colebow) [reply docs/src/main/sphinx/release/release-450.md:5]: I'm honestly not too sure what the feature is doing, and it wasn't documented, so there's not much to fall back on 😔

File: docs/src/main/sphinx/release/release-450.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
+# Release 450 (12 Jun 2024)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:1]: 18

File: docs/src/main/sphinx/release/release-450.md (hunk 9 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-450.md:5]: Should we reword around performance and make it higher level?

File: docs/src/main/sphinx/release/release-450.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  property. ({issue}`21375`)
+* {{breaking}} Improve performance of distinct aggregations, and replace the
+  `optimizer.mark-distinct-strategy` and
+  `optimizer.optimize-mixed-distinct-aggregations` configuration properties with
+  the new `optimizer.distinct-aggregations-strategy` property. ({issue}`21907`)
+* Improve performance of reading JSON files. ({issue}`22348`)
+* Improve performance for the {func}`date_trunc`, {func}`date_add`, and
+  {func}`date_diff` functions. ({issue}`22192`)
+* Fix failure when loading the [](/admin/event-listeners-openlineage). ({issue}`22228`)
+* Fix potential incorrect results when a cached value has been removed or
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:26]:
```suggestion
* Fix potential incorrect results when a cached value was removed or
```

File: docs/src/main/sphinx/release/release-450.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+## JDBC driver
+
+* Add support for the `assumeNullCatalogMeansCurrent` connection property. When
+  enabled, a `null` value for the `catalog` parameter in `DatabaseMetaData`
+  methods is assumed to mean the current catalog. If no current catalog is
+  set, the behaviour is unmodified. ({issue}`20866`)
+
+## BigQuery connector
+
+* Add support for name caching when the
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:49]: name of what?

File: docs/src/main/sphinx/release/release-450.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+
+## Hudi connector
+
+* Set the default value for the `s3.max-connections` configuration property
+  to 500. ({issue}`22209`)
+
+## Iceberg connector
+
+* Add support for reading Iceberg
+  [UniForm](https://docs.delta.io/latest/delta-uniform.html) tables. ({issue}`22106`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:94]: Is it really "Iceberg Uniform" or just "Uniform"

File: docs/src/main/sphinx/release/release-450.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+* Add support for the `TRUNCATE` statement. ({issue}`22340`)
+* Add support for V2 of the Nessie REST API. Previous behavior can be
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:96]: So its actually .. Change to use the V2 protocol by default..

File: docs/src/main/sphinx/release/release-450.md [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+  structured types. ({issue}`22347`)
+
+## MariaDB connector
+
+* Add support for [fault-tolerant execution](/admin/fault-tolerant-execution). ({issue}`22328`)
+* Improve performance of listing table columns. ({issue}`22241`)
+
+## Memory connector
+
+* Add support for the `TRUNCATE` statement. ({issue}`22337`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:113]: Nice!

File: docs/src/main/sphinx/release/release-450.md [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+
+## MySQL connector
+
+* Improve performance of listing table columns. ({issue}`22241`)
+
+## Pinot connector
+
+* Add support for the
+  [`enableNullHandling` query option](https://docs.pinot.apache.org/developers/advanced/null-value-support#advanced-null-handling-support). ({issue}`22214`)
+* Fix failure when using dynamic tables. ({issue}`22301`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-450.md:123]: Those are Pinot dynamic tables I assume ... should we link to some docs or so (I couldnt find any quickly)

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-450.md (hunk 14 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-450.md:96]: We didn't previously support V2, did we?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-450.md:95]: no .. but we are not only adding support .. we are switching as well..

Comment (mosabua) [reply docs/src/main/sphinx/release/release-450.md:95]: so maybe that is also a breaking change?

**Review by mosabua — APPROVED** [review]
One more thing to consider

File: docs/src/main/sphinx/release/release-450.md (hunk 9 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-450.md:5]: I think we should remove this item. It's too low level and doesn't have much direct visible impact to end users.

File: docs/src/main/sphinx/release/release-450.md [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+  property. ({issue}`22218`)
+* Expose driver execution statistics via JMX. ({issue}`22427`)
+* Improve performance of the {func}`first_value` and {func}`last_value`
+  functions. ({issue}`22092`)
+* Improve performance for large clusters under heavy workloads. ({issue}`22039`)
+* Improve performance of queries with simple predicates. This optimization can
+  be disabled using the `experimental.columnar-filter-evaluation.enabled`
+  configuration property or the `columnar_filter_evaluation_enabled` session
+  property. ({issue}`21375`)
+* {{breaking}} Improve performance of distinct aggregations, and replace the
```
Comment (martint) [inline docs/src/main/sphinx/release/release-450.md:19]: "... of aggregations containing a `DISTINCT` clause ..."

File: docs/src/main/sphinx/release/release-450.md [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+  `optimizer.mark-distinct-strategy` and
+  `optimizer.optimize-mixed-distinct-aggregations` configuration properties with
+  the new `optimizer.distinct-aggregations-strategy` property. ({issue}`21907`)
+* Improve performance of reading JSON files. ({issue}`22348`)
+* Improve performance for the {func}`date_trunc`, {func}`date_add`, and
+  {func}`date_diff` functions. ({issue}`22192`)
+* Fix failure when loading the [](/admin/event-listeners-openlineage). ({issue}`22228`)
+* Fix potential incorrect results when a cached value was removed or updated. ({issue}`22285`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-450.md:27]: Cached where? What values?

File: docs/src/main/sphinx/release/release-450.md [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+
+## Security
+
+* Add support for exclusively using web identity for authentication when using
```
Comment (martint) [inline docs/src/main/sphinx/release/release-450.md:31]: This item is hard to parse and understand. How about:

"Add support for using web identity exclusively for authentication when running on Amazon EKS with the legacy S3 file system enabled. This can be configured via the `trino.s3.use-web-identity-token-credentials-provider` property."

File: docs/src/main/sphinx/release/release-450.md [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+## BigQuery connector
+
+* Add support for metadata caching when the
+  `bigquery.case-insensitive-name-matching` configuration property is enabled. ({issue}`10740`)
+* {{breaking}} Automatically configure BigQuery scan parallelism, and remove the
+  `bigquery.parallelism` configuration property. ({issue}`22279`)
+
+## Cassandra connector
+
+* Fix incorrect behavior that uses one more partition than expected when
```
Comment (martint) [inline docs/src/main/sphinx/release/release-450.md:56]: What does "uses one more partition than expected mean"? Process a partition? Read a partition? Return results from a partition?

File: docs/src/main/sphinx/release/release-450.md (hunk 18 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-450.md:24]: In many different places, and it depends... metadata caches for some connectors, regular data caches for others.
